### ISSUE TITLE
`UnusedImports` can't detect unused `import java.util.*;` 

### ISSUE THREAD

**Issue by blindpirate** [checkstyle/checkstyle#7400]
Check documentation: https://checkstyle.org/config_xxxxxx.html#NameOfAffectedCheck

```
/var/tmp $ javac Main.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.1//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_1.dtd">
<module name="Checker">
    <property name="localeLanguage" value="en"/>
    <module name="TreeWalker">
        <module name="UnusedImports"/>
    </module>
</module>

/var/tmp $ cat YOUR_FILE.java
import java.util.*;

public class Main {
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Main.java
Starting audit...
Audit done.
```

---------------

Unused `java.util.*` imports should be detected by `UnusedImports` but they're not. I understand checkstyle is working on source-code-level so can't scan a package, but `java.util.*` is not a problem IMHO.

--------------

**Comment by blindpirate** [comment]
Just saw the doc

https://github.com/checkstyle/checkstyle/blob/179fe0d6dec566460a7c887ae0dcd708fb12bec6/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheck.java#L48-L50

So this issue would be more like a feature request.

**Comment by romani** [comment]
>Check documentation: https://checkstyle.org/config_xxxxxx.html#NameOfAffectedCheck

Why do you skip providing correct  link to Check documentation ?
Please let me know what can be improved to let author of issue provide link to documentation

**Comment by blindpirate** [comment]
Sorry @romani I added the link to the doc. This is documented behavior, but I'd like to make it a feature request.

**Comment by romani** [comment]
From documentation:

>An import statement is considered unused if:

It is not referenced in the file. The algorithm does not support wild-card imports like import java.io.*;. Most IDE's provide very sophisticated checks for imports that handle wild-card imports.

So technically we declared that it is not supported case.
Even your minimalist case (when only one wild-card import is used, or when all wildcard are not in use) can be supported. But it will not work in more real code examples. 
If there is a type and no specific import and there are few wildcard imports, checkstyle can not figure out from what wildcard imports it come from. So false negatives are not avoidable in case of wildcard imports.

@pbludov , @strkkk , @rnveach , should we support a case when all wildcard are not in use(when all types have specific import) for sure ?

**Comment by blindpirate** [comment]
I totally understand the current situation. Please feel free to close the issue or leave it as a feature that might be implemented in the future.
